add slope == 0 case into standard leaky relu nn test

glaringlee opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#37559 add slope == 0 case into standard leaky relu nn test**

Differential Revision: [D21319922](https://our.internmc.facebook.com/intern/diff/D21319922)

Commit:
add slope == 0 case into standard leaky relu nn test

[ghstack-poisoned]

albanD commented on test/test_autograd.py:
```diff
@@ -5635,11 +5635,6 @@ def test_leaky_relu_inplace_with_zero_or_neg_slope(self, device):
         with self.assertRaisesRegex(RuntimeError, "call out-of-place version"):
             b.backward(torch.ones(2, device=device))
 
-        a = torch.tensor([-2., 0., 2.], device=device, requires_grad=True)
```
I think this is worth keeping!
We do rely on this subgradient being 0. Even though we most likely will never willingly change that. It is good to have a test that makes sure it does not happen by mistake.

glaringlee replied:
sure. I will make a separate test to test this.

albanD reviewed (changes requested):
Thanks for the PR.
The new test looks good !

Commit:
Update on "add slope == 0 case into standard leaky relu nn test"


Differential Revision: [D21319922](https://our.internmc.facebook.com/intern/diff/D21319922)

[ghstack-poisoned]

albanD reviewed (approved):
Thanks!
